polaris-react/src/components/LegacyFilters/components/ConnectedFilterControl/ConnectedFilterControl.scss

```diff
@@ -59,6 +60,11 @@
 .RightContainer {
   display: flex;
 
+  /* Stop children of the filters from being overlapped due to buttons having a
+  * default min-width: 0.
+  * See more: https://stackoverflow.com/a/36247448 */
+  flex-shrink: 0;
```

Comment:
Only happens when the `children` prop is passed to `LegacyFilters`:

![overlap](https://github.com/Shopify/polaris/assets/612020/e089ac91-4427-45e1-821d-6de421a0eee0)

Disallowing shrinking fixes it:

![no-overlap](https://github.com/Shopify/polaris/assets/612020/25e1fc13-ce68-41f4-982c-6a986097c244)